tests/ui/needless_return.fixed

```diff
@@ -69,6 +69,27 @@ fn test_void_match(x: u32) {
     }
 }
 
+mod no_lint_if_stmt_borrows {
+    mod issue_5858 {
+        fn read_line() -> String {
+            use std::io::BufRead;
+            let stdin = ::std::io::stdin();
+            return stdin.lock().lines().next().unwrap().unwrap();
+        }
+
+        fn read_line2(value: bool) -> String {
```

Comment:
We should check that this is linted because the borrow does not happen in the last statement of the block. Removing the early return I mentioned in my previous comment will fix this.

We could maybe change the name to make the intent more clear, what about `borrows_but_not_last`?